mode/type/standalone/repository/provider/r2dbc-mariadb/src/main/java/org/apache/shardingsphere/mode/repository/standalone/r2dbc/mariadb/R2DBCMariaDBRepository.java

```diff
@@ -0,0 +1,178 @@
+import io.r2dbc.pool.ConnectionPool;
+import io.r2dbc.pool.ConnectionPoolConfiguration;
+import io.r2dbc.spi.Connection;
+import io.r2dbc.spi.ConnectionFactories;
+import io.r2dbc.spi.ConnectionFactory;
+import io.r2dbc.spi.ConnectionFactoryOptions;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shardingsphere.mode.repository.standalone.StandalonePersistRepository;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.UUID;
+
+/**
+ * R2DBC MariaDB repository.
+ */
+@Slf4j
+public class R2DBCMariaDBRepository implements StandalonePersistRepository {
+    
+    private static final String SEPARATOR = "/";
+    
+    private ConnectionPool connectionPool;
+    
+    @Override
+    public void init(final Properties props) {
+        ConnectionFactoryOptions connectionFactoryOptions = ConnectionFactoryOptions.builder()
+                .option(ConnectionFactoryOptions.DRIVER, "pool")
+                .option(ConnectionFactoryOptions.PROTOCOL, "mariadb")
+                .option(ConnectionFactoryOptions.HOST, props.getProperty("host"))
+                .option(ConnectionFactoryOptions.PORT, Integer.parseInt(props.getProperty("port")))
+                .option(ConnectionFactoryOptions.USER, props.getProperty("user"))
+                .option(ConnectionFactoryOptions.PASSWORD, props.getProperty("password"))
+                .option(ConnectionFactoryOptions.DATABASE, props.getProperty("database"))
+                .build();
+        ConnectionFactory connectionFactory = ConnectionFactories.get(connectionFactoryOptions);
+        ConnectionPoolConfiguration connectionPoolConfiguration = ConnectionPoolConfiguration.builder(connectionFactory).build();
+        connectionPool = new ConnectionPool(connectionPoolConfiguration);
+        Mono.usingWhen(connectionPool.create(),
+                connection -> Mono.from(connection.createBatch()
+                        .add("DROP TABLE IF EXISTS `repository`")
+                        .add("CREATE TABLE IF NOT EXISTS `repository`(id varchar(36) PRIMARY KEY, `key` TEXT, `value` TEXT, parent TEXT)")
+                        .execute()),
+                Connection::close)
+                .block();
+    }
+    
+    @Override
+    public String getDirectly(final String key) {
+        return Mono.usingWhen(connectionPool.create(),
+                connection -> Mono.from(connection.createStatement("SELECT `value` FROM `repository` WHERE `key` = ?")
+                        .bind(0, key)
+                        .execute())
+                        .flatMap(result -> Mono.from(result.map((row, rowMetadata) -> row.get("value", String.class))))
+                        .doOnError(throwable -> log.error("Get {} data by key: {} failed", getType(), key, throwable)),
+                Connection::close)
+                .block();
```

Comment:
If we have to block the thread, why not just use the standard JDBC driver?

Reply:
- Does this mean setting MySQL's JDBC driver to maven test scope to avoid GPL LICENSE issues? Or must we avoid using the GPL LICENSE? 

- I have checked the processing of the implementation of H2Database. Can testcontainer be used for regular tests? The init function seems to have to actually have a data source.

Reply:
About license, we have already introduced MySQL JDBC driver as test scope. Our code just using standard JDBC interface. MySQL JDBC driver could be introduced by user themselves, just like how ShardingSphere-Proxy distribution did.

Could we just add a `MySQLJDBCRepositoryProvider` which implements `org.apache.shardingsphere.mode.repository.standalone.jdbc.provider.JDBCRepositoryProvider`?

About the second question, I'm not familiar with it, sorry.

Reply:
- I looked back at what the original issue was trying to express, and I noticed that what he actually needed was something else. So I decided to delete the R2DBC implementation in the new commit.

- As for testcontainers, I am referring to the specific Docker container Url of `jdbc:tc:mysql:8.0.31:///config`.